test/unit/collections/CollectionsRepository.test.ts

```diff
@@ -0,0 +1,118 @@
+import { CollectionsRepository } from '../../../src/collections/infra/repositories/CollectionsRepository'
+import axios from 'axios'
+import {
+  ApiConfig,
+  DataverseApiAuthMechanism
+} from '../../../src/core/infra/repositories/ApiConfig'
+import {
+  createCollectionModel,
+  createCollectionPayload
+} from '../../testHelpers/collections/collectionHelper'
+import { TestConstants } from '../../testHelpers/TestConstants'
+import { ReadError } from '../../../src'
+
+describe('CollectionsRepository', () => {
+  const sut: CollectionsRepository = new CollectionsRepository()
+  const testCollectionSuccessfulResponse = {
+    data: {
+      status: 'OK',
+      data: createCollectionPayload()
+    }
+  }
+  const testCollectionModel = createCollectionModel()
+
+  beforeEach(() => {
+    ApiConfig.init(
+      TestConstants.TEST_API_URL,
+      DataverseApiAuthMechanism.API_KEY,
+      TestConstants.TEST_DUMMY_API_KEY
+    )
+  })
+  describe('getCollection', () => {
+    const expectedRequestConfigApiKey = {
+      params: {
+        returnOwners: true
+      },
+      headers: TestConstants.TEST_EXPECTED_AUTHENTICATED_REQUEST_CONFIG_API_KEY.headers
+    }
+
+    describe('by numeric id', () => {
+      test('should return Collection when providing a numeric id', async () => {
+        jest.spyOn(axios, 'get').mockResolvedValue(testCollectionSuccessfulResponse)
+        const expectedApiEndpoint = `${TestConstants.TEST_API_URL}/dataverses/${testCollectionModel.id}`
+
+        // API Key auth
+        let actual = await sut.getCollection(testCollectionModel.id)
+
+        expect(axios.get).toHaveBeenCalledWith(expectedApiEndpoint, expectedRequestConfigApiKey)
+        expect(actual).toStrictEqual(testCollectionModel)
+
+        actual = await sut.getCollection(testCollectionModel.id)
+        expect(axios.get).toHaveBeenCalledWith(expectedApiEndpoint, expectedRequestConfigApiKey)
+        expect(actual).toStrictEqual(testCollectionModel)
```

Comment:
The last 3 lines seem to be duplicated
```suggestion
        let actual = await sut.getCollection(testCollectionModel.id)

        expect(axios.get).toHaveBeenCalledWith(expectedApiEndpoint, expectedRequestConfigApiKey)
        expect(actual).toStrictEqual(testCollectionModel)
```